src/controller/expenses.js

```diff
@@ -0,0 +1,140 @@
+'use strict';
+
+const expensesServices = require('../services/expensesServices');
+const usersServices = require('../services/usersServices');
+
+const getAll = (req, res) => {
+  const {
+    userId,
+    categories,
+    from,
+    to,
+  } = req.query;
+
+  let expenses = expensesServices.getAll();
+
+  if (userId) {
+    expenses = expenses
+      .filter(expense => expense.userId === +userId);
+  }
+
+  if (from) {
+    expenses = expenses.filter(expense => (
+      new Date(expense.spentAt) >= new Date(from))
+    );
+  }
+
+  if (to) {
+    expenses = expenses.filter(expense => (
+      new Date(expense.spentAt) <= new Date(to))
+    );
+  }
+
+  if (categories) {
+    expenses = expenses
+      .filter(expense => categories.includes(expense.category));
+  }
+
+  res.statusCode = 200;
+  res.send(expenses);
+};
+
+const getById = (req, res) => {
+  const expenseId = +req.params.id;
```

Comment:
```suggestion
  const expenseId = Number(req.params.id);
```

The same in other places